service/src/main/java/tds/assessment/repositories/impl/StrandQueryRepositoryImpl.java

```diff
@@ -43,15 +55,88 @@ public StrandQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTemplate)
                 "   segments._key = :key";
 
         return new HashSet<>(jdbcTemplate.query(SQL, parameters, (rs, row) ->
-            new Strand.Builder()
-                .withName(rs.getString("name"))
-                .withKey(rs.getString("_key"))
-                .withMinItems(rs.getInt("minitems"))
-                .withMaxItems(rs.getInt("maxitems"))
-                // calling getObject() and casting to Float because .getFloat() defaults to 0 if null
-                .withAdaptiveCut((Float) rs.getObject("adaptivecut"))
-                .withSegmentKey(rs.getString("segmentKey"))
-                .build()
+            buildStrandFromResultSet(rs)
         ));
     }
+
+    @Override
+    public List<ContentLevelSpecification> findContentLevelSpeficationsBySegmentKey(final String segmentKey) {
+        SqlParameterSource parameters = new MapSqlParameterSource("segmentKey", segmentKey);
+
+        final String SQL = "(SELECT" +
+            "   S._fk_Strand AS contentLevel, \n" +
+            "   S.minItems, \n" +
+            "   S.maxItems, \n" +
+            "   S.isStrictMax, \n" +
+            "   S.bpweight, \n" +
+            "   S.adaptiveCut, \n" +
+            "   S.StartAbility, \n" +
+            "   S.StartInfo, \n" +
+            "   S.Scalar, \n" +
+            "   CASE WHEN S.StartAbility IS NOT NULL THEN 'true' ELSE 'false' END AS isReportingCategory, \n" +
+            "   S.abilityWeight, \n" +
+            "   S.precisionTarget, \n" +
+            "   S.precisionTargetMetWeight, \n" +
+            "   S.precisionTargetNotMetWeight, \n" +
+            "   CASE WHEN SS._fk_Parent IS NULL THEN 0 ELSE 1 END AS elementType \n" +
+            "FROM tbladminstrand S  \n " +
+            "JOIN tblstrand SS ON SS._Key = S._fk_Strand \n" +
+            "WHERE S._fk_AdminSubject = :segmentKey ) \n" +
+            "UNION ALL \n" +
+            "(SELECT \n" +
```

Comment:
@jeffjohnson9046 has run into issues with MySql 5.6 and sub-selects before, although I don't remember the details.  Just wanted to make sure this is gtg?